Core/Src/main.c

```diff
@@ -87,18 +89,53 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+    //Initializing buffers
+    uint8_t transmit_bits[3] = {0x0, 0x1, 0x80}; // populated with start and configure bits for CH0
+    uint8_t receive_bits[3] = {0x0};
+
+    //Initializing constants and variables
+    const uint16_t MAX_ADC_VALUE = 0x400 - 1; //as it's a 10 bit ADC (represents 00000011 11111111)
+    const uint16_t TOTAL_TIMER_COUNTS = 64000;
+    const uint16_t MIN_PERIOD = 0.05;
```

Comment:
So here you have declared a uint16_t which is only WHOLE positive numbers. If you want decimals, you would have to use float or double.